kanidmd/src/lib/logbuffer.rs

```diff
@@ -0,0 +1,68 @@
+use std::{io, ops};
+use tracing_subscriber::fmt::MakeWriter;
+
+// This type may be moved out of `kanidmd` later
+#[derive(Default)]
+pub struct LogBuffer {
+    buf: Vec<u8>,
+}
+
+// An "owned" type pointing to a buffer in a
+// `LogBuffer`. This type exists to satisfy the
+// `MakeWriter` trait restriction on `make_writer`,
+// which must return an owned `io::Write` object.
+// The benefit of using a raw ptr is that creating this
+// object of very cheap and fast, and unless `Subscriber`s
+// are opening a new thread of each log (I don't think they are),
+// this will never cross a thread boundry.
+pub struct LogBufferWriter(*mut Vec<u8>);
+
+impl ops::Drop for LogBuffer {
+    fn drop(&mut self) {
+        // logs are written to stderr when dropped
+        use io::Write;
+        #[allow(clippy::expect_used)]
+        io::stderr()
```

Comment:
In the future this will probably need to be sent to a queue for flushing to stderrl/stdout instead, rather than writing during drop.

Reply:
Agreed. I was aiming to have the least amount of code possible for this initial PR, and using `Drop` meant that I didn't have to add any explicit dump function to call before cleanup, but I'm happy to change this.